### ISSUE TITLE
Test XML output doesn't report passing test cases

### ISSUE THREAD

**Issue by bduffany** [bazelbuild/rules_go#2778]
### What version of rules_go are you using?

0.25.1

### What version of gazelle are you using?

0.22.3

### What version of Bazel are you using?

3.7.0

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

Linux / x86_64

### What did you do?

Run this go test:

```go
package test

import (
	"fmt"
	"testing"
)

func TestThatPasses(t *testing.T) {
	fmt.Println("TestThatPasses: Doing nothing.")
}

func TestThatFails(t *testing.T) {
	fmt.Println("TestThatFails: Calling t.FailNow()")
	t.FailNow()
}
```

### What did you expect to see?

Test XML output should show 2 test cases with 1 failed.

### What did you see instead?

Test XML output shows 1 test case with 1 failed:

```
<testsuites>
	<testsuite errors="0" failures="1" skipped="0" tests="1" time="0.002" name="github.com/buildbuddy-io/buildbuddy/server/test_environment">
		<testcase classname="test_environment" name="TestThatFails" time="0.000">
			<failure message="Failed" type="">--- FAIL: TestThatFails (0.00s)&#xA;</failure>
		</testcase>
	</testsuite>
</testsuites>
```

See the invocation here: https://app.buildbuddy.io/invocation/512f4d66-78a6-4e70-895e-6b9c5ea89a57?target=//server/test:go_default_test

**Comment by jayconrod** [comment]
cc @robbertvanginkel in case you have bandwidth

**Comment by aw185176** [comment]
@bduffany we got around this by using `--test_arg=-test.v`, which seems to force results to be produced even for tests that pass.
